pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator_helper.go

```diff
@@ -32,13 +33,19 @@ import (
 func (c *K8sOrchestrator) getPVCAnnotations(ctx context.Context, volumeID string) (map[string]string, error) {
 	log := logger.GetLogger(ctx)
 	log.Debugf("Getting annotations on pvc corresponding to volume: %s", volumeID)
+	pvcAnn := make(map[string]string)
 	if pvc := c.volumeIDToPvcMap.get(volumeID); pvc != "" {
```

Comment:
Would we ever have a case where the map does not have the volume?

Reply:
We might have a case during container restart. 
This map gets initialized during container init, and all existing cluster volumes are added to the map using informer's event handlers. 
https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/94c57fe8b0bd89e8f72ce8687d7bb118f374b255/pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go#L328

So when a pod, PVC are deleted - if PVC gets deleted before it's processed by event handler during restart, and the detach call is processed after container restart, the map will not have the volume.
Since this again happens when a PVC has been deleted, I have added to return `NotFound` error if a volume is not in the map which will be handled in `ClearFakeAttach`